queue/util/message-parser.js

```diff
@@ -1,9 +1,22 @@
 const msgpack = require('msgpack-lite')
+const Model = require('~/models/Model')
+const Event = require('~/models/events/Event')
 
+/**
+* Encode an object or a fashion-model
+* @param message {Object|Model}
+* @return {Buffer} - Encoded message using msgpack
+*/
 exports.encode = function (message) {
-  return msgpack.encode(message)
+  let data = Model.isModel(message) ? message.clean() : message
+  return msgpack.encode(data)
 }
 
+/**
+* Decodes a message and wraps in an Event model
+* @param message {Buffer} - Buffer to decode and wrap into an Event model
+*/
 exports.decode = function (message) {
-  return msgpack.decode(message)
+  message = msgpack.decode(message)
+  return Event.wrap(message)
```

Comment:
Just so the others know, the `convertData` is the responsibility of the service since errors should be logged and handled appropriately.